tests/runner.py

```diff
@@ -70,6 +72,11 @@ def is_emterpreter(self):
 
   def setUp(self):
     Settings.reset()
+
+    for root, dirnames, filenames in os.walk(TEMP_DIR):
```

Comment:
Well, if it's only the test runner leaking, i'm not too worried - we can clean the bot directories manually too, like you suggested here. But we should have tests on emcc not leaking, if we don't already.

Another worry with this pull is false positives, if another process adds a temp file meanwhile.

Overall I understand your concern, and it bugs me too. But I'm not sure what is a good solution, except to check and find leaks manually and fix them.